jdaviz/configs/imviz/plugins/coords_info/coords_info.py

```diff
@@ -562,8 +562,9 @@ def _copy_axes_to_spectral():
                 if '_pixel_scale_factor' in sp.meta:
                     eqv = [(u.MJy / u.sr,
                             u.MJy,
-                            lambda x: (x * sp.meta['_pixel_scale_factor']),
+                            lambda x: (x * np.asarray(sp.meta.get('_pixel_scale_factor', 1))),
                             lambda x: x)]
+                    eqv += u.spectral_density(sp.spectral_axis)
```

Comment:
I cannot leave code suggestion here because of GitHub limitation but I think this whole `if` block needs to be re-written.

First, you import this at the top of the module:

```python
from jdaviz.utils import _eqv_pixar_sr
```

Then the `if` block should be like this (also, `get` is unnecessary because the `if` already checked if the meta keyword exists):

```python
if '_pixel_scale_factor' in sp.meta:
    eqv = u.spectral_density(sp.spectral_axis) + _eqv_pixar_sr(sp.meta['_pixel_scale_factor'])
```

Reply:
I did use this update with `_eqv_pixar_sr` in the `marks.py` diff yesterday (I will update any other spots I've touched with this conditional so it's standardized). There are a lot of places where we need this now so it is worth pinging everyone else with tickets in the sprint relating to unit conversion know to use this too.